diff -upr XQilla-2.3.4.old/include/xqilla/framework/XQillaExport.hpp XQilla-2.3.4/include/xqilla/framework/XQillaExport.hpp
--- XQilla-2.3.4.old/include/xqilla/framework/XQillaExport.hpp	2018-09-05 20:16:40 +0000
+++ XQilla-2.3.4/include/xqilla/framework/XQillaExport.hpp	2018-09-05 20:35:53 +0000
@@ -48,7 +48,7 @@
  * XQILLA_API functions as being imported from a DLL, wheras this DLL
  * sees symbols defined with this macro as being exported.
  */
-#if defined(WIN32) && !defined(__CYGWIN__) && !defined(__MINGW__)
+#if defined(WIN32) && !defined(__CYGWIN__) && !defined(__MINGW32__)
 #  if !defined(XQILLA_API)
 #    if defined(XQILLA_APIS)
 #      define XQILLA_API __declspec(dllexport)
diff -upr XQilla-2.3.4.old/src/functions/FunctionSerializeJSON.cpp XQilla-2.3.4/src/functions/FunctionSerializeJSON.cpp
--- XQilla-2.3.4.old/src/functions/FunctionSerializeJSON.cpp	2018-09-05 20:16:41 +0000
+++ XQilla-2.3.4/src/functions/FunctionSerializeJSON.cpp	2018-09-05 20:36:10 +0000
@@ -20,7 +20,7 @@
 #include "../config/xqilla_config.h"
 
 #include <stdio.h> // for sprintf
-#if defined(WIN32) && !defined(__CYGWIN__) && !defined(__MINGW__)
+#if defined(WIN32) && !defined(__CYGWIN__) && !defined(__MINGW32__)
 #define snprintf _snprintf
 #endif
 
diff -upr XQilla-2.3.4.old/src/utils/DateUtils.cpp XQilla-2.3.4/src/utils/DateUtils.cpp
--- XQilla-2.3.4.old/src/utils/DateUtils.cpp	2018-09-05 20:16:44 +0000
+++ XQilla-2.3.4/src/utils/DateUtils.cpp	2018-09-06 10:12:43 +0000
@@ -22,7 +22,7 @@
 #include <math.h>
 #include <climits>
 #include <stdio.h> // for sprintf
-#if defined(WIN32) && !defined(__CYGWIN__) && !defined(__MINGW__)
+#if defined(WIN32) && !defined(__CYGWIN__) && !defined(__MINGW32__)
 #define snprintf _snprintf
 #endif
 
diff -upr XQilla-2.3.4.old/src/yajl/yajl_common.h XQilla-2.3.4/src/yajl/yajl_common.h
--- XQilla-2.3.4.old/src/yajl/yajl_common.h	2018-09-05 20:16:42 +0000
+++ XQilla-2.3.4/src/yajl/yajl_common.h	2018-09-05 20:38:07 +0000
@@ -38,7 +38,7 @@
 /* msft dll export gunk.  To build a DLL on windows, you
  * must define WIN32, YAJL_SHARED, and YAJL_BUILD.  To use a shared
  * DLL, you must define YAJL_SHARED and WIN32 */
-#if defined(WIN32) && defined(YAJL_SHARED)
+#if defined(WIN32) && defined(YAJL_SHARED) && !defined(__CYGWIN__)  && !defined(__MINGW32__)
 #  ifdef YAJL_BUILD
 #    define YAJL_API __declspec(dllexport)
 #  else
diff -upr XQilla-2.3.4.old/tests/xmark/xmark.cpp XQilla-2.3.4/tests/xmark/xmark.cpp
--- XQilla-2.3.4.old/tests/xmark/xmark.cpp	2018-09-05 20:16:38 +0000
+++ XQilla-2.3.4/tests/xmark/xmark.cpp	2018-09-06 09:52:35 +0000
@@ -67,7 +67,7 @@ string timestamp(string &dateTime)
   time(&tt);
 
   struct tm *tm_p;
-#ifdef _MSC_VER   
+#if defined(_MSC_VER) || defined(__MINGW32__)
   tm_p = localtime(&tt);
 #else
   struct tm tm;
